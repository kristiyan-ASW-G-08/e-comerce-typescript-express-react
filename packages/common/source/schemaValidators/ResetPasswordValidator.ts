import * as yup from 'yup';

interface UserValidatorType {
  password: string;
  confirmPassword: string;
}
const UserValidator = yup.object<UserValidatorType>().shape({
  confirmPassword: yup
    .string()
    .trim()
    .min(12)
    .oneOf([yup.ref('password')], "Passwords don't match")
    .required(),
});

export default UserValidator;
