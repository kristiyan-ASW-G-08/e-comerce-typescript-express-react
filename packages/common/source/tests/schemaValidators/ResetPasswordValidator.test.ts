import ResetPasswordValidator from '@schemaValidators/ResetPasswordValidator';

describe('ResetPasswordValidator', (): void => {
  const validPasswords = ['123456789012', 'somePassword10', 'validPassword'];
  const invalidPasswords = [
    { password: 'somePassword10', confirmPassword: 'validPassword' },
    { password: '' },
    { password: 'Password' },
  ];
  it.each(validPasswords)('should validate successfully', async password => {
    await expect(
      ResetPasswordValidator.validate(
        { password, confirmPassword: password },
        { abortEarly: false },
      ),
    ).resolves.toStrictEqual({ password, confirmPassword: password });
  });
  it.each(invalidPasswords)('should validate unsuccessfully', async body => {
    await expect(
      ResetPasswordValidator.validate(body, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
