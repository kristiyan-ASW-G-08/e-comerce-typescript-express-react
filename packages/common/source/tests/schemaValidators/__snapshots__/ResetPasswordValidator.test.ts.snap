// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordValidator should validate unsuccessfully 1`] = `[ValidationError: Passwords don't match]`;

exports[`ResetPasswordValidator should validate unsuccessfully 2`] = `[ValidationError: confirmPassword is a required field]`;

exports[`ResetPasswordValidator should validate unsuccessfully 3`] = `[ValidationError: confirmPassword is a required field]`;
